component/plugincontroller/component.go

```diff
@@ -0,0 +1,34 @@
+package plugincontroller
+
+import (
+	"github.com/docker/docker/component"
+	"github.com/docker/docker/plugin"
+	"golang.org/x/net/context"
+)
+
+const name = "plugincontroller"
+
+// Set sets the plugin controller component
+func Set(s *plugin.Manager) (cancel func(), err error) {
+	return component.Register(name, s)
+}
+
+// Wait waits for the plugin controller component to be available
+func Wait(ctx context.Context) *plugin.Manager {
+	c := component.Wait(ctx, name)
+	if c == nil {
+		return nil
+	}
+
+	// This could panic... but I think this is ok.
+	// This should never be anything else
+	return c.(*plugin.Manager)
+}
+
+func Get() *plugin.Manager {
+	c := component.Get(name)
```

Comment:
Why this abstraction instead of getting the component from a registry directly?

It seems like the value of a component registry is that you don't need to be aware of where that component is coming from. If you are getting the component directly from a package we've lost that value.

So either we could remove the component registry, and just pass around references, or we should drop these helpers.